node/node_test.go

```diff
@@ -841,30 +830,39 @@ func TestGenesisConfig(t *testing.T) {
 		app.Config.DataDirParent = t.TempDir()
 
 		require.NoError(t, app.Initialize())
+		t.Cleanup(func() { app.Cleanup(context.Background()) })
+
 		var existing config.GenesisConfig
 		require.NoError(t, existing.LoadFromFile(filepath.Join(app.Config.DataDir(), genesisFileName)))
 		require.Empty(t, existing.Diff(app.Config.Genesis))
 	})
+
 	t.Run("no error if no diff", func(t *testing.T) {
 		app := New()
 		app.Config = getTestDefaultConfig(t)
 		app.Config.DataDirParent = t.TempDir()
 
 		require.NoError(t, app.Initialize())
+		t.Cleanup(func() { app.Cleanup(context.Background()) })
+
 		app.Cleanup(context.Background())
 		require.NoError(t, app.Initialize())
```

Comment:
I had to think long why you call `t.Cleanup(func() { app.Cleanup(context.Background()) })` and immediately cleanup the app one line below. I suggest instead:
```suggestion
		require.NoError(t, app.Initialize())
		app.Cleanup(context.Background())

		require.NoError(t, app.Initialize())
		app.Cleanup(context.Background())
```